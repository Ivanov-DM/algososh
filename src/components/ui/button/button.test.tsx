import React from 'react';
import renderer from 'react-test-renderer';
import {render, screen, fireEvent} from '@testing-library/react';

import { Button } from './button';

function handleClick() {
    alert('Button was pressed')
}

describe('Tests for button component', () => {
    it('Button renders correctly with attribute text', () => {
        const tree = renderer
            .create(<Button text="Тест"/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Button renders correctly without attribute text', () => {
        const tree = renderer
            .create(<Button/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Button renders correctly with attribute disabled', () => {
        const tree = renderer
            .create(<Button text="" disabled={true}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Button renders correctly with attribute isLoader', () => {
        const tree = renderer
            .create(<Button text="" isLoader={true}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Callback is called correctly when the button is pressed', () => {
        window.alert = jest.fn();
        render(<Button text="Test" onClick={handleClick}/>)
        const button = screen.getByText('Test');
        fireEvent.click(button);
        expect(window.alert).toHaveBeenCalledWith('Button was pressed');
    });
})