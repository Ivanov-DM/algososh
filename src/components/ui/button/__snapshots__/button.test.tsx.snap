// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for button component Button renders correctly with attribute disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    
  </p>
</button>
`;

exports[`Tests for button component Button renders correctly with attribute isLoader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Tests for button component Button renders correctly with attribute text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Тест
  </p>
</button>
`;

exports[`Tests for button component Button renders correctly without attribute text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
